src/main/java/com/groupby/tracker/GbTracker.java

```diff
@@ -226,6 +237,34 @@ private String capitalize(String s) {
         }
     }
 
+    private List<Metadata> setSiteFilterMetadataValue(List<Metadata> metadata)
+    {
+        if (metadata == null)
+        {
+            metadata = new ArrayList<>();
+        }
+
+        boolean valueSet = false;
+        for (int i = 0; i < metadata.size(); i++)
+        {
+            Metadata siteFilterMetadata = metadata.get(i);
+            if (siteFilterMetadata.getKey().equalsIgnoreCase("siteFilter"))
+            {
+                valueSet = true;
+            }
+        }
+
+        if (!valueSet)

```

Comment:
This should check if !valueSet && this.siteFilterMetadataValue != null before adding the value. We don't want to add the metadata value if it has not been explicitly set to anything; but we want to preserve if they explicitly set it to "".